app/components/submission/WithCurrentSubmission.js

```diff
@@ -89,42 +89,47 @@ const UPDATE_SUBMISSION = gql`
 `
 
 class WithCurrentSubmission extends React.Component {
-  constructor() {
-    super()
-    this.state = { error: null, data: null, loading: true }
-  }
+  state = { error: null, data: null, loading: true }
+
   componentDidMount() {
-    this.props.client
-      .query({ query: GET_CURRENT_SUBMISSION })
-      .then(({ data }) => {
-        if (data.currentSubmission) {
-          return this.setState({ data, error: null, loading: false })
-        }
-        return this.props.client
-          .mutate({
-            mutation: CREATE_SUBMISSION,
-            refetchQueries: [
-              {
-                query: GET_CURRENT_SUBMISSION,
-              },
-            ],
-          })
-          .then(({ data: createData }) => {
-            this.setState({
-              data: createData,
-              error: undefined,
-              loading: false,
-            })
-          })
-      })
-      .catch(error => this.setState({ error, data: null, loading: false }))
+    this.querySubscription = this.props.client
+      .watchQuery({ query: GET_CURRENT_SUBMISSION })
+      .subscribe(
+        ({ data }) => {
+          if (data.currentSubmission) {
+            this.setData(data)
+          } else {
+            this.props.client
+              .mutate({
+                mutation: CREATE_SUBMISSION,
+                refetchQueries: [{ query: GET_CURRENT_SUBMISSION }],
+              })
+              .then(result => this.setData(result.data))
+              .catch(error => this.setError(error))
+          }
+        },
+        error => this.setError(error),
+      )
+  }
+
+  setData(data) {
+    this.setState({ data, error: undefined, loading: false })
```

Comment:
Out of curiosity, why do we need `error: undefined` here, rather than `error: null` like we have in the initial state?

Reply:
Good point. I don't think we do.

Reply:
Is this being changed before we merge?

Reply:
No?

I'm not saying it should have been. Just unsure why it's like this? :woman_shrugging:

Reply:
Ah, sorry! Should have been, yeah. Will change when I'm next there.